src/Altinn.App.Core/Helpers/UserHelper.cs

```diff
@@ -81,25 +69,40 @@ public async Task<UserContext> GetUserContext(HttpContext context)
         UserProfile userProfile =
             await _profileClient.GetUserProfile(userContext.UserId)
             ?? throw new Exception("Could not get user profile while getting user context");
+
         userContext.UserParty = userProfile.Party;
 
-        if (context.Request.Cookies[_settings.GetAltinnPartyCookieName] != null)
-        {
-            userContext.PartyId = Convert.ToInt32(
-                context.Request.Cookies[_settings.GetAltinnPartyCookieName],
-                CultureInfo.InvariantCulture
-            );
-        }
+        userContext.PartyId = cookieValue is not null
+            ? Convert.ToInt32(cookieValue, CultureInfo.InvariantCulture)
+            : userContext.PartyId;
 
-        if (userContext.PartyId == userProfile.PartyId)
+        userContext.Party = userContext.PartyId.Equals(userProfile.Party?.PartyId)
+            ? userContext.Party = userProfile.Party
+            : await _altinnPartyClientService.GetParty(userContext.PartyId);
+
+        userContext.SocialSecurityNumber = userContext.Party?.SSN ?? userContext.UserParty.SSN;
+
+        return userContext;
+    }
+
+    private static ClaimWrapper GetClaim(IEnumerable<Claim> claims, string claimType)
+    {
+        var claim = claims.FirstOrDefault(x => x.Type.Equals(claimType, StringComparison.Ordinal))?.Value;
+        return new ClaimWrapper(claim);
+    }
+
+    private readonly record struct ClaimWrapper(string? Value)
+    {
+        public static implicit operator string?(ClaimWrapper claimWrapper)
         {
-            userContext.Party = userProfile.Party;
+            return claimWrapper.Value;
         }
-        else
+
+        public static implicit operator int(ClaimWrapper claimWrapper)
         {
-            userContext.Party = await _altinnPartyClientService.GetParty(userContext.PartyId);
+            return string.IsNullOrEmpty(claimWrapper.Value)
+                ? default
+                : Convert.ToInt32(claimWrapper.Value, CultureInfo.InvariantCulture);
```

Comment:
When the underlying operation is fallible, I think it's nicer to have the conversion be explicit. In other words I'm not sure `ClaimWrapper` is worth it (I understand that this changes nothing functionally, but it puts more distance between the knowledge of the value and the fallible operation)

Reply:
How about something like `ClaimWrapper.TryParseInt()` or something similar? If we aren't using an intermediary object, we have to go back to checking for nulls (eg. the value wasnt found), which means something along the lines of the original structure.

I mean, it clearly worked, so perhaps this was some unnecessary abstraction on my part.

Reply:
Can't we have the same behavior just without the intermediary? E.g. a local function

```csharp
private static T GetClaimOrDefault(string claimType, T defaultValue);
```

Then the caller can decide what the expectation is from the claim. 
Also, not a big deal considering the number of claims, but with this approach we are going from 1 full loop over the claims to 1 loop over the claims per type we want to find, so it should be a little less efficient